sql/src/main/java/io/crate/action/sql/Session.java

```diff
@@ -272,9 +272,8 @@ public void bind(String portalName,
             throw t;
         }
 
-        AnalyzedStatement unboundStatement = preparedStmt.unboundStatement();
         AnalyzedStatement analyzedStatement;
-        if (unboundStatement == null) {
+        if (preparedStmt.analyzedStatement() == null) {
```

Comment:
Do you think it would make sense to now always eagerly analyze the statement within `parse` already? That would allow us to remove the special case here and in `describe`. And we could probably also get rid of the `analyzedStatement` setter on `PreparedStmt`.

Reply:
makes sense! 👍 I'll push a commit that does that to this PR.